sqlglot/dialects/mysql.py

```diff
@@ -555,11 +555,22 @@ class Generator(generator.Generator):
             exp.WeekOfYear: rename_func("WEEKOFYEAR"),
         }
 
-        TYPE_MAPPING = generator.Generator.TYPE_MAPPING.copy()
-        TYPE_MAPPING.pop(exp.DataType.Type.MEDIUMTEXT)
-        TYPE_MAPPING.pop(exp.DataType.Type.LONGTEXT)
-        TYPE_MAPPING.pop(exp.DataType.Type.MEDIUMBLOB)
-        TYPE_MAPPING.pop(exp.DataType.Type.LONGBLOB)
+        TYPE_MAPPING = {
+            **{
+                k: v
+                for k, v in generator.Generator.TYPE_MAPPING.items()
+                if k
+                not in {
+                    exp.DataType.Type.MEDIUMTEXT,
+                    exp.DataType.Type.LONGTEXT,
+                    exp.DataType.Type.MEDIUMBLOB,
+                    exp.DataType.Type.LONGBLOB,
+                }
+            },
+            exp.DataType.Type.TIMESTAMP: "DATETIME",
+            exp.DataType.Type.TIMESTAMPTZ: "TIMESTAMP",
+            exp.DataType.Type.TIMESTAMPLTZ: "TIMESTAMP",
```

Comment:
I think we can do something similar to what I did [here](https://github.com/tobymao/sqlglot/pull/2172/files#diff-14ef9d8b99f48dcf8f6a4e3bbab7bf55a9d85d2b83d684c3c822f0f94fcd1d07R558) to factor this types out.

Reply:
Ah I see you just changed that. I didn't see the code change when I copy/pasted this over. Will fix.